polymath/extensions/vector_ops.py

```
@@ -134,7 +137,7 @@ def _check_axis(arg, axis, op):
         try:
             _ = selections[i]
         except IndexError:
-            raise IndexError(f'axis is out of range ({-arg._rank},{arg._rank}) in '
+            raise IndexError(f'axis is out of range ({-arg._ndims},{arg._ndims}) in '
                              f'{type(arg)}.{op}: {i}')
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)